cmd/load-runner/helm.go

```diff
@@ -66,12 +88,84 @@ func (h *helmClient) Uninstall(config *parser.LoadTestConfig) error {
 	return h.executor.run(context.Background(), "helm", args...)
 }
 
-func processHelmEnvVars(args []string, envVars map[string]string) []string {
+func processHelmEnvVars(args []string, envVars map[string]string, logger logger.Logger) []string {
+	args = calculateLoadParameters(args, envVars, logger)
 	for key, value := range envVars {
 		if strings.Contains(value, ",") {
 			value = strings.ReplaceAll(value, ",", "\\,")
 		}
 		args = append(args, "--set", fmt.Sprintf("deployment.env.%s=%s", key, value))
 	}
+
+	return args
+}
+
+func calculateLoadParameters(args []string, envVars map[string]string, logger logger.Logger) []string {
+	resourceCalculation := envVars["RESOURCE_CALCULATION"]
+
+	// If no resource calculation is specified, return args unchanged
+	if resourceCalculation == "" {
+		return args
+	}
+
+	// Validate resource calculation value
+	if resourceCalculation != "auto" && !strings.HasPrefix(resourceCalculation, "overprovision,") {
+		logger.Warn("Invalid RESOURCE_CALCULATION value",
+			"value", resourceCalculation,
+			"expected", "auto or overprovision,<percentage>")
+		return args
+	}
+
+	const (
+		baseEventsPerResourceUnit = 5000
+
+		concurrencyUnit = 2000
+
+		messageGeneratorsUnit = 500
+
+		cpuUnit = 1
+
+		memoryUnit = 2
+	)
+
+	maxEventsPerSecond, err := strconv.Atoi(envVars["MAX_EVENTS_PER_SECOND"])
+	if err != nil {
+		logger.Fatal("Failed to convert MAX_EVENTS_PER_SECOND to int: %v", err)
```

Comment:
`Fataln` for non-sugared (I hope I'll be able to provide a linter soon, bear with me :D)

Reply:
There are other occurrences here and there in this file, won't comment on each one of course.